From 5f669c209d293752ef103146d594b7bd6c125f5d Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Mon, 8 Sep 2014 15:08:02 -0400
Subject: [ethernet] ixgbevf: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate

Message-id: <1410188883-6595-48-git-send-email-jogreene@redhat.com>
Patchwork-id: 92366
O-Subject: [RHEL7.1 PATCH 47/48] [ixgbevf] net: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate
Bugzilla: 1091124
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

BZ: 1091124
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7919168
Tested: In beaker. See BZ for details

[redhat: ixgbevf filtered]
upstream commit 537fae0101c8853abb52136788173dde74b9d1e9

Signed-off-by: Jiri Pirko <jiri@resnulli.us>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/ixgbevf/ethtool.c b/drivers/net/ethernet/intel/ixgbevf/ethtool.c
index ddd9b50..79219fc 100644
--- a/drivers/net/ethernet/intel/ixgbevf/ethtool.c
+++ b/drivers/net/ethernet/intel/ixgbevf/ethtool.c
@@ -135,8 +135,8 @@ static int ixgbevf_get_settings(struct net_device *netdev,
   ethtool_cmd_speed_set(ecmd, speed);
   ecmd->duplex = DUPLEX_FULL;
  } else {
-  ethtool_cmd_speed_set(ecmd, -1);
-  ecmd->duplex = -1;
+  ethtool_cmd_speed_set(ecmd, SPEED_UNKNOWN);
+  ecmd->duplex = DUPLEX_UNKNOWN;
  }
 
  return 0;
-- 
1.7.1